### PR TITLE
folder structure change for i18n

### PR DESCRIPTION
**Pull request by SamYuan1990**
enable i18n settings in `config.toml`
folder structure change for i18n, moving all english content to en folder.

### CODE REVIEW

**Comment by SamYuan1990** [comment]
preview at https://deploy-preview-271--tag-env-sustainability.netlify.app/zh/landscape/

**Comment by SamYuan1990** [comment]
@rootfs , @leonardpahlke could you please help with this PR?

**Comment by leonardpahlke** [comment]
Will take a look at it tomorrow

**Comment by leonardpahlke** [comment]
cc @cjyabraham @chalin

**Review by cjyabraham — CHANGES_REQUESTED** [review]
Adding another translation adds the language dropdown to the topnav. This requires space and now at certain browser widths the topnav is wrapping to two lines:

Screenshot 2023-11-14 at 4 19 14 PM

Is it possible to shorten "Cloud Native Sustainability Week" to just "Sustainability Week"? If not, we'll need to tweak the breakpoints so we can avoid it looking bad.

Also, are you planning on having multiple translations of the entire site? If so, you may want to organize the translations by directory, as we've done on [the glossary](https://github.com/cncf/glossary/tree/main/content).

**Comment by chalin** [comment]
Drive by comments ... 

- IMHO, "Cloud Native Sustainability Week" should be under events. By moving it, that would resolve the top-nav spacing issue :)
- Given the maintenance overhead, for other projects, we usually want folks to commit to maintaining language translations for a while before bring it in.
- Also, there are some Korean pages. If this site starts officially using the multilingual feature, then the Korean pages should be "upgraded" to us that features.

**Review by chalin — COMMENTED** [review]
Drive by comments ... 

- IMHO, "Cloud Native Sustainability Week" should be under events. By moving it, that would resolve the top-nav spacing issue :)
- Given the maintenance overhead, for other projects, we usually want folks to commit to maintaining language translations for a while before bring it in.
- Also, there are some Korean pages. If this site starts officially using the multilingual feature, then the Korean pages should be "upgraded" to us that features.

**Comment by chalin** [comment]
Also, there is a way to set this up so that any given language whose page translations are missing fall back to `en`.

**Comment by cjyabraham** [comment]
> > IMHO, "Cloud Native Sustainability Week" should be under events. By moving it, that would resolve the top-nav spacing issue :)
> 
> Lets do that. We put it on the front page since folks were searching for this, and this lowered the barrier to find the resource. Now the week is over, we can move it. Is there a way to perseve the link? tag-env-sustainability.cncf.io/cloud-native-sustainability-week/ after we moved the page?

You can use [an alias in the frontmatter of the page](https://bwaycer.github.io/hugo_tutorial.hugo/extras/aliases/).

**Comment by cjyabraham** [comment]
> Also, there is a way to set this up so that any given language whose page translations are missing fall back to `en`.

I think that might be a strange UX. It would mean someone who deliberately chooses Chinese goes instead to the English page. 

Currently if a page is missing and someone chooses Chinese, it takes them to the homepage of the site in Chinese, which I think makes more sense (assume we translate the homepage). At that point the user can see in the nav what pages are available in that language.

**Comment by SamYuan1990** [comment]
Thanks @leonardpahlke , @cjyabraham and @chalin 
Sorry for replay late. I am quite busy in recently, and as I contribute to open source in my personal time...
I suppose we have items here as todo:
-  We need to move the week into event, and keep alias ref https://bwaycer.github.io/hugo_tutorial.hugo/extras/aliases/
for this, I suppose Leo can open a ticket. (let's see who has time to make it, as thanks giving is near by)
- I do agree that we need to contribute into specific folder's for different language. Which means:
  1. Sustainable computing is a global things, hence we need i18n as we don't want language becomes blocker between our communication.
  1. Technologically we will need to restructure our folder with EN for English?
  1. For our audience, it will look strange as just one or two pages in native language and rest of them in English.
for this, I suppose we need one task for EN folder, and a kind of PR template to enable i18n for a language. (which means i18n team should contribute to native language content 1st then enable the i18n option on website?)

For this PR, maybe I should close it?
it will be a good place as our discussion for i18n related things, and others can reference later on?

**Comment by leonardpahlke** [comment]
issue opened: https://github.com/cncf/tag-env-sustainability/issues/278

**Comment by SamYuan1990** [comment]
@leonardpahlke , I try to follow @cjyabraham 's suggestion create an en folder and move the contents. But there some lint issue. I suppose I need to add skip lint in those files?

**Comment by SamYuan1990** [comment]
@guidemetothemoon @leonardpahlke , I don't mind to ref https://github.com/cncf/tag-env-sustainability/commit/7d314130232afcaba82b55f340dbcb33bbc5210e#diff-6dbf3e19d9e5049af0f8b9e1727da1c04a899297cf6a835bdde0e0ad7b3d62d7R54 to add technological wordings. but we still find people's name in our meet up page. Hence is there any way to skip `cSpell` checking? as there will be people's name and adding people's name as glossary is too wired. 
any other suggestion?

**Comment by leonardpahlke** [comment]
Thanks @SamYuan1990 for working on this!! 🙌 

> Sustainable computing is a global things, hence we need i18n as we don't want language becomes blocker between our communication.

yes, in the last meeting we also had voices about adding content in Spanish. I am sure there are also folks that would like to add content in German and other languages. +1 adding multi language support across the website and follow the [glossary folder structure](https://github.com/cncf/glossary/tree/main/content) to achive this, as previously proposed. We can update the folder structure with this PR. Governance wise, we could discuss a content translator role in the TAG (but we will do that outside of this PR). 

> I don't mind to ref https://github.com/cncf/tag-env-sustainability/commit/7d314130232afcaba82b55f340dbcb33bbc5210e#diff-6dbf3e19d9e5049af0f8b9e1727da1c04a899297cf6a835bdde0e0ad7b3d62d7R54 to add technological wordings like Istio. but we still find people's name in our meet up page. Hence is there any way to skip cSpell checking? as there will be people's name and adding people's name as glossary is too wired.
any other suggestion?

@guidemetothemoon perhaps you can answer this one best.

**Comment by SamYuan1990** [comment]
> @leonardpahlke @SamYuan1990 We've recently refactored QA checks for pull requests, where spelling validation hasn't properly worked until now... but now it does, which is the reason for why you get more errors on modified files. I have a plan to run all those validations towards existing files in the repo to just fix most of the issues for what already exists in the repo, but I haven't come that far yet.
> 
> Regarding the names: it doesn't look like cSpell is very good at recognizing names and there's no easy, straightforward way to ignore those so I'm currently looking into ways to resolve this scenario in a more efficient manner. Since I'm planning to run these validations towards existing files in the repo in the coming days I would say it's ok just to merge the changes with the failing spelling check since those will be fixed anyway in a subsequent PR. Or if the PR will still be active for some days then those will be resolved by themselves once I publish the PR that will resolve these issues.
> 
> P.S. the tool did raise some meaningful errors that would be great to update if you will be doing any further updates in this PR (f.ex. "timewindow" or "manufactural") 😊

Option one: keep this PR open and wait @guidemetothemoon to check all existing wording and then rebase.
Option two: I am fix this PR with all missing wording. No matter it's name or .... but I can fix typo or keep everything to Istio. Then @guidemetothemoon check all the files later on?
May I know we are on same page, as if both of us make correction on files, we will conflicts aren't we?

**Comment by guidemetothemoon** [comment]
@SamYuan1990 almost what I was thinking, but to avoid conflicts I can suggest following:
- If the QA checks is the only blocker for merging this PR then we can merge it as is and I will fix the raised QA errors in the subsequent PR;
or
- You can resolve the errors as part of this PR and temporary add the names to our custom dictionary (basically option two that you mentioned), as explained here: https://github.com/cncf/tag-env-sustainability/blob/main/writing-style.md#fixing-qa-errors . Once I add a more efficient solution for handling the spell check of names I will clean up what's needed in a subsequent PR, after this PR is merged, to avoid conflicts;

I hope it made it a bit clearer? What do you think?

**Comment by leonardpahlke** [comment]
Sounds good to me thanks @SamYuan1990 :)

**Review by cjyabraham — CHANGES_REQUESTED** [review]
It's a bit hard to follow what this PR is trying to do without reading through all the comments. Perhaps keeping the PR name and description up-to-date would be helpful for anyone trying to review.

I have read through the comments, however, so it seems that this PR is for moving all files into /en/, which works fine. Could you remove the website/content/en/landscape/_index.zh.md file, since that's not the correct filename format anyway, right? I assume that was just leftover from previous commits.

**Comment by guidemetothemoon** [comment]
@SamYuan1990 I've published a PR which contains further improvements to the QA checks and establishes the baseline for existing content: #289

Once it's merged the amount of build errors in your PR should also be reduced. I've also tried to handle the spell check of names as optimal as possible, but it's a challenging string to filter completely so in some cases you may still need to add an ignore field to filter out the specific name, as described here: https://github.com/cncf/tag-env-sustainability/blob/ebee70989abac87b8f8b5df405a4a0bbd0869c88/writing-style.md

If you still experience any errors after #289 is merged, please let me know and I can add a change suggestions directly on your PR to fix it asap.

Thanks!

**Comment by SamYuan1990** [comment]
> It's a bit hard to follow what this PR is trying to do without reading through all the comments. Perhaps keeping the PR name and description up-to-date would be helpful for anyone trying to review.
> 
> I have read through the comments, however, so it seems that this PR is for moving all files into /en/, which works fine. Could you remove the website/content/en/landscape/_index.zh.md file, since that's not the correct filename format anyway, right? I assume that was just leftover from previous commits.

but if so, btw @leonardpahlke , @cjyabraham may I know any comments for Korean content? Honestly I can't speak Korean, should I make a Korean folder? so I can just move existing content in to a Korean folder.  Or others can make it in another pr? after this PR been merged?

**Comment by leonardpahlke** [comment]
Yes, if we have content in korean already, lets create the folder. FYI @sysnet4admin @seokho-son

File: link-config.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "aliveStatusCodes": [200, 206, 402],
+  "aliveStatusCodes": [200, 206, 400, 402, 403],
```
Comment (guidemetothemoon) [inline link-config.json:2]:
```suggestion
  "aliveStatusCodes": [200, 206, 402],
```

In most of the cases 400 and 403 is a sign for bad link and we don't want to mark those codes as OK. I understand that you added this due to the few links that the build failed on, but I have explained in detail why those links failed here: https://github.com/cncf/tag-env-sustainability/pull/289#issuecomment-1828103998

Comment (guidemetothemoon) [reply link-config.json:2]: You may ignore the link check build error as part of your PR - it will be fixed once #289 is merged.

**Review by guidemetothemoon — CHANGES_REQUESTED** [review]

**Comment by cjyabraham** [comment]
> > Yes, if we have content in korean already, lets create the folder. FYI @sysnet4admin @seokho-son
> 
> please let me know if a ko folder is needed and nice to cover in this PR. @cjyabraham

Personally, I don't see the need for an empty ko folder in this PR. Any team who submits ko translations can include that in their PR. Either way is fine, however, and I'd prioritize getting this PR merged.

**Review by cjyabraham — APPROVED** [review]

**Comment by leonardpahlke** [comment]
Thanks!
